src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseAccessGroupUtils.java

```diff
@@ -0,0 +1,232 @@
+package uk.gov.hmcts.ccd.domain.service.common;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.hmcts.ccd.domain.model.definition.AccessTypeRoleDefinition;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseAccessGroup;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseAccessGroupWithId;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.endpoint.exceptions.ValidationException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.UUID;
+
+public class CaseAccessGroupUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CaseAccessGroupUtils.class);
+    public static final String CASE_ACCESS_GROUPS = "CaseAccessGroups";
+    private static final String CCD_ALL_CASES = "CCD:all-cases-access";
+
+    private static final String ORGANISATION = "Organisation";
+    private static final String ORGANISATIONID = "OrganisationID";
+    private static final String ORG_IDENTIFIER_TEMPLATE = "$ORGID$";
+    private static final String ORG_POLICY_CASE_ASSIGNED_ROLE = "OrgPolicyCaseAssignedRole";
+    private static final String CASE_ACCESS_GROUP_TYPE = "caseAccessGroupType";
+    private static final String GROUPACCESS_VALUE = "value";
+    private static final String GROUPACCESS_ID = "id";
+
+    private CaseDataService caseDataService;
+
+    public void updateCaseAccessGroupsInCaseDetails(CaseDetails caseDetails,
```

Comment:
I would expect to see a test class for this given how much logic we have in here.

Reply:
The SubmitCaseTransactionCaseAccessGroupTest.java tests this and the coverage shows all logic is covered.